lib/src/platform/src/push_channel_native.dart

```diff
@@ -0,0 +1,83 @@
+import '../../common/common.dart';
+import '../../error/error.dart';
+import '../../generated/platform_constants.dart';
+import '../../push_notifications/push_notifications.dart';
+import '../../realtime/realtime.dart';
+import '../../rest/rest.dart';
+import '../platform.dart';
+
+/// The native code implementation of [PushChannel].
+class PushChannelNative extends PlatformObject implements PushChannel {
+  final String _name;
+
+  /// A rest client used platform side to invoke push notification methods
+  final RestInterface? rest;
+
+  /// A realtime client used platform side to invoke push notification methods
+  final RealtimeInterface? realtime;
+
+  /// Pass the channel name and an Ably realtime or rest client.
+  PushChannelNative(this._name, {this.rest, this.realtime}) {
+    final ablyClientNotPresent = rest == null && realtime == null;
+    final moreThanOneAblyClientPresent = rest != null && realtime != null;
+    if (ablyClientNotPresent || moreThanOneAblyClientPresent) {
+      throw Exception(
+          'Specify one Ably client when creating ${(PushNative).toString()}.');
+    }
+  }
+
+  @override
+  Future<int?> createPlatformInstance() => (realtime != null)
+      ? (realtime as Realtime).handle
+      : (rest as Rest).handle;
+
+  /// Subscribe device to the channel’s push notifications.
+  @override
+  Future<void> subscribeDevice() => invoke(
+      PlatformMethod.pushSubscribeDevice, {TxTransportKeys.channelName: _name});
+
+  /// Unsubscribe device from the channel’s push notifications.
+  @override
+  Future<void> unsubscribeDevice() => invoke(
+      PlatformMethod.pushUnsubscribeDevice,
+      {TxTransportKeys.channelName: _name});
+
+  @override
+  Future<void> subscribeClient() => invoke(
+      PlatformMethod.pushSubscribeClient, {TxTransportKeys.channelName: _name});
+
+  @override
+  Future<void> unsubscribeClient() => invoke(
+      PlatformMethod.pushUnsubscribeClient,
+      {TxTransportKeys.channelName: _name});
+
+  @override
+  Future<PaginatedResultInterface<PushChannelSubscription>> listSubscriptions(
+      Map<String, String> params) async {
+    if (!params.containsKey('deviceId') &&
+        !params.containsKey('clientId') &&
+        !params.containsKey('deviceClientId') &&
+        !params.containsKey('channel')) {
+      // This error only happen on Android. They are thrown here
+      // for both platforms (iOS/ Android) to make the API more consistent.
+      final errorInfo = ErrorInfo(
+          code: 40000,
+          href: 'https://help.ably.io/error/40000',
+          message:
+              "expected parameter 'deviceId', 'clientId', 'deviceClientId', and/or 'channel'");
+
+      throw AblyException(
+          '40000',
+          "expected parameter 'deviceId', 'clientId', 'deviceClientId', and/or 'channel'",
+          errorInfo);
+    }
```

Comment:
Good idea, done in e53a5f915e139d1fb09c21323ae41157600c434d

Reply:
href should be dynamic; based on the error code, I believe. Can you check the error code href documentation for better impelementation?

Reply:
href should be dynamic; based on the error code, I believe. Can you check the error code href documentation for better implementation?

Reply:
Oh i see, and also, the AblyException code is different to the ErrorInfo code. Im going to leave the AblyException `null` because they are different, WDYT? d0fdbf92e3e8c47c8922cfc8ab3713b61534c145

Reply:
Actually, after discussing on slack, that extra convenience constructor was not necessary. Removed in 89a9176cf1ee33a88048cd45771995ddc6278357